java/src/Ice/src/main/java/com/zeroc/Ice/DNSException.java

```diff
@@ -1,60 +1,25 @@
-//
-// Copyright (c) ZeroC, Inc. All rights reserved.
-//
-//
-// Ice version 3.7.10
-//
-// <auto-generated>
-//
-// Generated from file `LocalException.ice'
-//
-// Warning: do not edit this file.
-//
-// </auto-generated>
-//
+// Copyright (c) ZeroC, Inc.
 
 package com.zeroc.Ice;
 
-/**
- * This exception indicates a DNS problem. For details on the cause, {@link DNSException#error}
- * should be inspected.
- */
-public class DNSException extends LocalException {
-  public DNSException() {
-    this.error = 0;
-    this.host = "";
-  }
-
-  public DNSException(Throwable cause) {
-    super(cause);
-    this.error = 0;
-    this.host = "";
-  }
-
-  public DNSException(int error, String host) {
-    this.error = error;
+/** This exception indicates a DNS problem. */
+public final class DNSException extends SyscallException {
+  public DNSException(String host) {
+    super("Cannot resolve host '" + host + "'");
     this.host = host;
```

Comment:
Possibly surprising, but calling `super(message, null)` is slightly different than `super(message)`. since these constructors do different things, and the default value for `cause` isn't actually `null`, it's `self`, which acts as a special sentinel value.

My understanding is that explicitly setting a `null` cause is used to indicate that _something_ caused it, but for some reason it's unknown. Which is not the case here.